File: benchmarks/pandas/bench_dataframe_cumsum.py
import pandas as pd, time, json
N = 100_000
cols = 4
data = {f"col{c}": [(i % 10) + 1 for i in range(N)] for c in range(cols)}
df = pd.DataFrame(data)
WARMUP = 3
ITERS = 20
for _ in range(WARMUP):
    df.cumsum()
t0 = time.perf_counter()
for _ in range(ITERS):
    df.cumsum()
total = (time.perf_counter() - t0) * 1000
print(json.dumps({"function": "dataframe_cumsum", "mean_ms": total / ITERS, "iterations": ITERS, "total_ms": total}))

File: benchmarks/tsb/bench_dataframe_cumsum.ts
import { dataFrameCumsum } from "tsb";
import { DataFrame } from "tsb";
const N = 100_000;
const cols = 4;
const data: Record<string, number[]> = {};
for (let c = 0; c < cols; c++) {
  data[`col${c}`] = Array.from({ length: N }, (_, i) => (i % 10) + 1);
}
const df = new DataFrame(data);
const WARMUP = 3;
const ITERS = 20;
for (let i = 0; i < WARMUP; i++) dataFrameCumsum(df);
const t0 = performance.now();
for (let i = 0; i < ITERS; i++) dataFrameCumsum(df);
const total = performance.now() - t0;
console.log(JSON.stringify({ function: "dataframe_cumsum", mean_ms: total / ITERS, iterations: ITERS, total_ms: total }));
